src/plugins/index_pattern_field_editor/public/components/preview/field_preview_context.tsx

```diff
@@ -0,0 +1,495 @@
+            body: {
+              size: 1,
+              query: {
+                ids: {
+                  values: [id],
+                },
+              },
+            },
+          },
+        })
+        .toPromise()
+        .then((res) => [res, null])
+        .catch((err) => [null, err]);
+
+      setIsFetchingDocument(false);
+
+      let loadedDocuments: EsDocument[] = [];
+
+      if (response) {
+        if (response.rawResponse.hits.total > 0) {
+          setPreviewResponse({ fields: [], error: null });
+          loadedDocuments = response.rawResponse.hits.hits;
+        } else {
+          setPreviewResponse({
+            fields: [],
+            error: {
+              code: 'DOC_NOT_FOUND',
+              error: {
+                message: i18n.translate(
+                  'indexPatternFieldEditor.fieldPreview.error.documentNotFoundDescription',
+                  {
+                    defaultMessage:
+                      'Error previewing the field as the document provided was not found.',
+                  }
+                ),
+              },
+            },
+          });
+        }
+      } else if (error) {
+        // TODO: improve this error handling when there is a server
+        // error fetching a document
+        setPreviewResponse({
+          fields: [],
+          error: {
+            code: 'ERR_FETCHING_DOC',
+            error: {
+              message: error.toString(),
+            },
+          },
+        });
+      }
+
+      setClusterData({
+        documents: loadedDocuments,
+        currentIdx: 0,
+      });
+    },
+    [indexPattern, search]
+  );
+
+  const updatePreview = useCallback(async () => {
+    if (fieldTypeToProcess !== 'runtime' || !allParamsDefined()) {
+      return;
+    }
+
+    const currentApiCall = ++previewCount.current;
+
+    const response = await getFieldPreview({
+      index: currentDocIndex,
+      document: params.document!,
+      context: `${params.type!}_field` as FieldPreviewContext,
+      script: params.script!,
+    });
+
+    if (currentApiCall !== previewCount.current) {
+      // Discard this response as there is another one inflight
+      // or we have called reset() and don't need the response anymore.
+      return;
+    }
+
+    setIsLoadingPreview(false);
+
+    const { error: serverError } = response;
+
+    if (serverError) {
+      // Server error (not an ES error)
+      const title = i18n.translate('indexPatternFieldEditor.fieldPreview.errorTitle', {
+        defaultMessage: 'Failed to load field preview',
+      });
+      notifications.toasts.addError(serverError, { title });
+
+      return;
+    }
+
+    const data = response.data ?? { values: [], error: {} };
+    const { values, error } = data;
+
+    if (error) {
+      const fallBackError = {
+        message: i18n.translate('indexPatternFieldEditor.fieldPreview.defaultErrorTitle', {
+          defaultMessage: 'Error executing the script.',
```

Comment:
```suggestion
          defaultMessage: 'Unable to run the provided script',
```

Reply:
@sebelga, when would this error display? If Kibana hits an issue internally and can't run the script?

Reply:
Every time we execute the script in ES and there is an error. This is the title above the error returned by ES.